File: 14-longest-common-prefix/longest-common-prefix.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        i=0
        prefix = []
        
        n = math.inf
        for s in strs:
            n = min(n,len(s))
        
        while i<n:

            prefix.append(strs[0][i])

            for s in strs:
                if s[i]!=prefix[-1]:
                    prefix.pop()
                    return "".join(prefix)
            i+=1
        
        return "".join(prefix)

File: 14-longest-common-prefix/longest-common-prefix.js
/**
 * @param {string[]} strs
 * @return {string}
 */
var longestCommonPrefix = function(strs) {
    let n = Infinity
    let res = []

    for (let i = 0; i<strs.length;i++){
        if (strs[i].length<n){
            n = strs[i].length
        }
    }

    for (let i = 0 ; i<n; i++){
        res.push(strs[0][i])
        for (let j = 0;j<strs.length;j++){
            if (strs[j][i] == res[res.length-1]){
                continue
            }
            else{
                res.pop()
                return res.join("")
            }
        }
    }


    return res.join("")
};